mindsdb/integrations/handlers/financial_modeling_prep_handler/financial_modeling_handler.py

```diff
@@ -0,0 +1,103 @@
+import pandas as pd
+from typing import Dict
+from urllib.parse import urlencode
+from mindsdb.integrations.libs.api_handler import APIHandler
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+from mindsdb.integrations.handlers.financial_modeling_prep_handler.financial_modeling_tables import FinancialModelingTradesTable
+
+from urllib.request import urlopen
+from mindsdb.utilities import log
+import json
+import requests
+_FINANCIAL_MODELING_URL = 'https://financialmodelingprep.com/api/v3/'
+
+logger = log.getLogger(__name__)
+
+class FinancialModelingHandler(APIHandler):
+
+    name = "financial_modeling"
+
+    def __init__(self, name, connection_data: dict,  **kwargs):
+        super().__init__(name)
+
+        self.api_key = None
+        self.connection_data = connection_data
+        if "api_key" not in connection_data:
+            raise Exception(
+                "FINANCIAL_MODELING engine requires an API key. Retrieve an API key from https://site.financialmodelingprep.com/developer. See financial_modeling_prep_handler/README.MD on how to include API key in query."
+            )
+        self.api_key = connection_data['api_key']
+        self.client = None
+        self.is_connected = False
+
+        daily_chart_table = FinancialModelingTradesTable(self) 
+        self._register_table('daily_chart_table', daily_chart_table)
+
+    def connect(self) -> StatusResponse: 
```

Comment:
The `connect()` method of our API handlers is typically expected to return an instance of a client (package) that will allow us to interact with the API in question. Not an instance of `StatusResponse` (although this is incorrectly used as a type hint in some of our handlers). [Here](https://github.com/mindsdb/mindsdb/blob/bc102f4011837d289bca43d7978717f408818e6b/mindsdb/integrations/handlers/ms_teams_handler/ms_teams_handler.py#L55) is a good example of a correct implementation. Now, I understand that since you are using `requests` here, there is no client instance as such that can be returned. Therefore, you can choose to return something else; for instance the `base_url` and then you can use this within your `BaseResultsTable` like `base_url = self.handler.connect()`. It is completely up to you how you want to handle this, but in essence, what needs to be done is to remove the `StatusResponse` logic from `connect()`.

Reply:
Makes sense! I adjusted the connect() method to return the base_url instead of a StatusResponse instance.